src/NServiceBus.Core.Tests/Transports/PushRuntimeSettingsTests.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace NServiceBus.Core.Tests.Transports
+{
+    using System;
+    using NUnit.Framework;
+    using Transport;
+
+    [TestFixture]
+    public class PushRuntimeSettingsTests
+    {
+        [Test]
+        public void Should_default_concurrency_to_num_processors()
+        {
+            Assert.AreEqual(Environment.ProcessorCount, new PushRuntimeSettings().MaxConcurrency);
```

Comment:
Do we care if this test will fail if it runs on a machine with a single logical processor? Do we ever run tests in VMs that could be configured that way?